File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/motion/safe_mode.cpp
// control_power.cpp

#include <iostream>
#include <iomanip>
#include <thread>
#include <chrono>
#include "h10-w/h10w_motion.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 上半身设置安全模式示例 ===" << std::endl;
  H10wMotion motion;

  std::cout << "正在进行上半身安全模式设置..." << std::endl;
  bool enable = true;
  if (motion.setSafeMode(true) == 0 && motion.getSafeMode(enable) == 0 &&
      enable == true) {
    std::cout << "上半身进入安全模式成功" << std::endl;
  } else {
    std::cerr << "上半身进入安全模式失败" << std::endl;
  }

  std::this_thread::sleep_for(std::chrono::seconds(3));

  if (motion.setSafeMode(false) == 0 && motion.getSafeMode(enable) == 0 &&
      enable == false) {
    std::cout << "上半身退出安全模式成功" << std::endl;
  } else {
    std::cerr << "上半身退出安全模式失败" << std::endl;
  }

  return 0;
}

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/motion/safe_mode.py
import humanoid_sdk_py
from humanoid_sdk_py import h10w
import time

h10w_motion = h10w.H10wMotion()
time.sleep(1)

print("=== 上半身设置安全模式示例 ===")

print("正在进行上半身安全模式设置...")
ret_s = h10w_motion.setSafeMode(True)
ret_g, enable = h10w_motion.getSafeMode()
if ret_s == 0 and ret_g == 0 and enable == True:
    print("上半身进入安全模式成功")
else:
    print("上半身进入安全模式失败")

time.sleep(1)

print("正在进行上半身安全模式设置...")
ret_s = h10w_motion.setSafeMode(False)
ret_g, policy = h10w_motion.getSafeMode()
if ret_s == 0 and ret_g == 0 and policy == False:
    print("上半身退出安全模式成功")
else:
    print("上半身退出安全模式失败")
